osutil/user/getent.go

```diff
@@ -0,0 +1,193 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package user
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+func getEnt(params ...string) ([]byte, error) {
+	cmd := exec.Command("getent", params...)
+	cmd.Stdin = nil
+
+	outBuf, err := cmd.Output()
+	if err != nil {
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
+			return nil, fmt.Errorf("getent returned an error: %q", exitError.Stderr)
+		}
+		return nil, fmt.Errorf("getent could not be executed: %w", err)
+	}
+
+	return outBuf, nil
+}
+
+// lookupFromGetent calls getent, parses and filters its output
+// The component at `index` will need to match `expectedValue`.
+// If `isKey`, then `expectedValue` will also be passed as parameter
+// to getent along `database`. `numComponents` should be 4 for groups
+// and 7 for users.
```

Comment:
would be a bit cleaner to use a map  database -> #fields and return an internal error if database is not in the map